src/controllers/legacycontrollermappingfilehandler.cpp

```diff
@@ -230,17 +269,109 @@ void LegacyControllerMappingFileHandler::addScriptFilesToMapping(
+        }
         scriptFile = scriptFile.nextSiblingElement("file");
     }
+
+#ifdef MIXXX_USE_QML
+    // Build a list of QML files to load
+    QDomElement screen = controller.firstChildElement("screens")
+                                 .firstChildElement("screen");
+
+    // Look for additional ones
+    while (!screen.isNull()) {
+        QString identifier = screen.attribute("identifier", "");
+        uint targetFps = screen.attribute("targetFps", "30").toUInt();
+        QString pixelFormatName = screen.attribute("pixelType", "RBG");
+        QString endianName = screen.attribute("endian", "little");
+        QString reversedColor = screen.attribute("reversed", "false").toLower();
+        QString rawData = screen.attribute("raw", "false").toLower();
+        uint splashOff = screen.attribute("splashoff", "0").toUInt();
+
+        if (!targetFps || targetFps > s_maxTargetFps) {
+            qWarning() << "Invalid target FPS. Target FPS must be between 1 and" << s_maxTargetFps;
+            return;
+        }
+
+        if (splashOff > s_maxSplashOffDuration) {
+            qWarning() << QString(
+                    "Invalid splashoff duration. Splashoff duration must be "
+                    "between 0 and %1. Clamping to %2")
+                                  .arg(s_maxSplashOffDuration)
+                                  .arg(s_maxSplashOffDuration);
+            splashOff = s_maxSplashOffDuration;
+        }
+
+        if (!kSupportedPixelFormat.contains(pixelFormatName)) {
+            qWarning() << "Unsupported pixel format" << pixelFormatName;
+            return;
+        }
+
+        if (!kEndianFormat.contains(endianName)) {
+            qWarning() << "Unknown endiant format" << endianName;
+            return;
+        }
+
+        QImage::Format pixelFormat = kSupportedPixelFormat.value(pixelFormatName);
+        std::endian endian = kEndianFormat.value(endianName);
+
+        uint width = screen.attribute("width", "0").toUInt();
+        uint height = screen.attribute("height", "0").toUInt();
+
+        if (!width || !height) {
+            qWarning() << "Invalid screen size. Screen size must have a width "
+                          "and height above 1 pixel";
+            return;
+        }
+
+        qDebug() << "Adding screen " << identifier;
+        mapping->addScreenInfo(identifier,
+                QSize(width, height),
+                targetFps,
+                splashOff,
+                pixelFormat,
+                endian,
+                reversedColor == "yes" || reversedColor == "true" || reversedColor == "1",
+                rawData == "yes" || rawData == "true" || rawData == "1");
+        screen = screen.nextSiblingElement("screen");
+    }
+    // Build a list of QML files to load
+    QDomElement qmlLibrary = controller.firstChildElement("qmllibraries")
+                                     .firstChildElement("library");
+
+    // Look for additional ones
+    while (!qmlLibrary.isNull()) {
```

Comment:
need an escape from this possible infinite loop

Reply:
`qmlLibrary` is updated with the next sibling at each iteration, or a null element is none found. This is the same pattern that was used [above ](https://github.com/mixxxdj/mixxx/blob/10cfd2fdc33d79dc8534a22748f3be7aaed841e4/src/controllers/legacycontrollermappingfilehandler.cpp#L315).  I'm not sure how I could change that,m any suggestion?

Reply:
ahh I see.  Nah it's ok as-is